yapapi/executor/__init__.py

```diff
@@ -255,6 +286,31 @@ async def find_offers() -> None:
                         emit(events.ProposalRejected(prop_id=proposal.id))
                     elif not proposal.is_draft:
                         try:
+                            prov_platforms = {
+                                property.split(".")[4]
+                                for property in proposal.props
+                                if property.startswith("golem.com.payment.platform.")
+                            }
+                            if not prov_platforms:
+                                prov_platforms = {"NGNT"}
+                            req_platforms = {
+                                allocation.payment_platform
+                                for allocation in self._budget_allocations
+                            }
+                            common_platforms = req_platforms.intersection(prov_platforms)
+                            if len(common_platforms) > 0:
+                                builder.properties["golem.com.payment.chosen-platform"] = next(
+                                    iter(common_platforms)
+                                )
+                            else:
+                                # reject proposal if there are no common payment platforms
+                                with contextlib.suppress(Exception):
+                                    await proposal.reject()
+                                emit(
+                                    events.ProposalRejected(
+                                        prop_id=proposal.id, reason="No common payment platforms"
+                                    )
+                                )
```

Comment:
Extract this whole new fragment of code to a separate method/function?

Reply:
Moved some code to `_get_common_payment_platforms`.